src/js/messaging/containers/Main.jsx

```diff
@@ -31,11 +40,13 @@ Main.propTypes = {
 const mapStateToProps = (state) => {
   return {
     folders: state.folders.data.items,
-    navExpanded: state.folders.ui.nav.expanded
+    navExpanded: state.folders.ui.nav.expanded,
+    modals: state.modals
```

Comment:
Pare it down to the create folder modal or its visibility since that's all you need in this component. `createFolderModal` or something.